### PR TITLE
Bats->InSpec, fixes, and updates

### PR DESCRIPTION
**Pull request by cheeseplus**
- Replace Bats tests with InSpec
- Fix builds for Fedora and openSUSE
- Exclude Fedora and openSUSE from ESL as there aren't repos for these
- Update source version to latest `19.2`

Signed-off-by: Seth Thomas <sthomas@chef.io>

### Description

[Describe what this change achieves]

### Issues Resolved

[List any existing issues this PR resolves]

### Check List

- [x] All tests pass. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/TESTING.MD>
- [x] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable
- [x] All commits have been signed for the Developer Certificate of Origin. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/CONTRIBUTING.MD>

### CODE REVIEW

File: .kitchen.docker.yml [hunk 1]
```diff
@@ -90,9 +89,9 @@ platforms:
     intermediate_instructions:
       - RUN zypper --non-interactive install aaa_base perl-Getopt-Long-Descriptive which net-tools
 
-- name: opensuse-42.1
+- name: opensuse-42.2
```
Comment (tas50) [inline .kitchen.docker.yml]: We can set this to -leap now which will probably save time updating it everytime they ship a release

File: recipes/esl.rb [hunk 2]
```diff
@@ -21,6 +21,8 @@
 
 case node['platform_family']
 when 'debian'
+  package 'apt-transport-https'
```
Comment (tas50) [inline recipes/esl.rb]: Is this to work around the docker images not having this? It *should* be on every distro install in the real world

**Comment by tas50** [comment]
Nice to see Fedora/Suse getting some love here

File: recipes/esl.rb (hunk 2 above)
Comment (cheeseplus) [inline recipes/esl.rb]: This was necessary to get tests passing and didn't seem to be part of the base images. I figured pulling it in would be a no-op for most things but still catch when it's not as well.

**Comment by tas50** [comment]
🚢

**Pull request merged**